### PR TITLE
feat: add simple per plugin exec metric

### PR DESCRIPTION
**Pull request by pasteley**
This PR adds per-plugin prometheus metrics to daemon mode.

Metrics are collected at cni executor boundary, so they cover each plugin in a conflist independently. Labels are deliberately bounded to avoid high-cardinality series from pod, namespace, NAD, interface etc. 

Described in docs/metrics.md

## Summary by CodeRabbit

* **New Features**
  * Added Prometheus metrics for CNI plugin lookups, execution outcomes, and execution duration.
  * Metrics include plugin names, commands, result labels, and daemon HTTP request counts.
  * Metrics are disabled by default and can be enabled through the daemon’s `metricsPort` configuration.
  * Added documentation for configuring and using the Multus daemon metrics endpoint.

* **Bug Fixes**
  * Improved metrics handling when the CNI command is unavailable by reporting it as `UNKNOWN`.

### CODE REVIEW

File: pkg/server/metrics_exec.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+func cniCommand(environ []string) string {
+	for _, entry := range environ {
+		command, found := strings.CutPrefix(entry, "CNI_COMMAND=")
+		if found {
+			return command
+		}
+	}
+	return unknownCNICommand
```
[the remark this answers is not in the document]
Comment (pasteley) [reply pkg/server/metrics_exec.go:94]: What's the idea behind validating over hardcoded list of commands here?
Why should we add `VERSION` there?

File: docs/metrics.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+| ---------------------------------------------- | ----------------------------- |
+| `multus_cni_plugin_lookup_total`               | CNI plugin lookup attempts    |
+| `multus_cni_plugin_execution_total`            | CNI plugin execution attempts |
+| `multus_cni_plugin_execution_duration_seconds` | CNI plugin execution duration |
+
+| Label     | Values                          | Meaning          |
+| --------- | ------------------------------- | ---------------- |
+| `plugin`  | CNI plugin type                 | CNI plugin       |
+| `command` | `ADD`, `DEL`, `CHECK`, `STATUS` | CNI command      |
+| `outcome` | `success`, `failure`            | Operation result |
```
[the remark this answers is not in the document]
Comment (pasteley) [reply docs/metrics.md]: Removed field explanations at all

**Comment by thomasferrandiz** [comment]
Thanks for the PR.

In asking Claude for a review, it found the following issue:

###  Bug: Test assertion checks wrong label value (will fail)
**File:** `pkg/server/metrics_exec_test.go` ~line 73/85
**Problem:** The third `ExecPlugin` call passes `[]string{"CNI_COMMAND=unsupported"}` as environ. `cniCommand()` sees the `CNI_COMMAND=` prefix and returns `"unsupported"` — **not** `"UNKNOWN"`. The `"UNKNOWN"` path is only hit when `CNI_COMMAND` is absent entirely.
The assertion then checks label `("sriov", "UNKNOWN", "success")` which was never incremented → counter is `0.0`, assertion expects `1.0` → **test fails**.

**Fix options:**
1. Change the third call's environ to `[]string{}` (empty) so `cniCommand()` actually returns `"UNKNOWN"` — this also provides real coverage of the missing-command edge case.
2. Change the assertion to check `"unsupported"` instead of `unknownCNICommand`.
Option 1 is preferable since it preserves the intended test coverage of the unknown-command path.

**Comment by thomasferrandiz** [comment]
Also the failing test was fixed in the master branch so you can rebase to get the fix.
Otherwise the PR looks good.

**Comment by pasteley** [comment]
@thomasferrandiz fixed, ptal
